src/lib.rs

```diff
@@ -726,7 +726,11 @@ impl Library {
                         continue;
                     }
 
-                    if statik && is_static_available(val, &system_roots, &dirs) {
+                    if val.starts_with(':') {
+                        // Pass this flag to linker directly.
+                        let meta = format!("cargo:rustc-link-arg={}{}", flag, val);
```

Comment:
This now would do `-l:foo.a` with a leading `:`. From what you wrote in the issue

> `Syntax: -l [KIND[:MODIFIERS]=]NAME[:RENAME].`

this doesn't seem valid, or am I missing something? You'd have an empty name then

Reply:
cargo:rustc-link-arg will pass the flags to ld directly, this flag is not be handled by rustc.

Reply:
-l:xxx link to a file named xxx, but there is no way to tell rustc to link to a file directly, if you pass -lxxx to rustc, rustc will try to link to libxxx.a instead , which is not what -l:xxx want

Reply:
Ah I see, that makes sense